pkg/kubelet/prober/prober_manager_test.go

```diff
@@ -299,13 +308,14 @@ func TestUpdatePodStatus(t *testing.T) {
 	m.UpdatePodStatus(testPodUID, &podStatus)
 
 	expectedReadiness := map[probeKey]bool{
-		{testPodUID, unprobed.Name, readiness}:              true,
-		{testPodUID, probedReady.Name, readiness}:           true,
-		{testPodUID, probedPending.Name, readiness}:         false,
-		{testPodUID, probedUnready.Name, readiness}:         false,
-		{testPodUID, notStartedNoReadiness.Name, readiness}: false,
-		{testPodUID, startedNoReadiness.Name, readiness}:    true,
-		{testPodUID, terminated.Name, readiness}:            false,
+		{testPodUID, unprobed.Name, readiness}:                        true,
+		{testPodUID, probedReady.Name, readiness}:                     true,
+		{testPodUID, probedPending.Name, readiness}:                   false,
+		{testPodUID, probedUnready.Name, readiness}:                   false,
+		{testPodUID, notStartedNoReadiness.Name, readiness}:           false,
+		{testPodUID, startedNoReadiness.Name, readiness}:              true,
+		{testPodUID, terminated.Name, readiness}:                      false,
+		{testPodUID, notStartedWithoutReadinessProbe.Name, readiness}: false,
```

Comment:
On line 307 we need to add a fake result with readinessManager where the cached result was previously `ready` and a container that has not been started yet flips the cached Ready result to `false`.

Reply:
I don't think we need to do this, if the state of container is Waiting,  if the state of a container is Waiting, the var started will become false in the fust judgment 
```
if c.State.Running == nil {
			started = false
} else
```
so , the following code wouldn't obtain the status from readinessmanager.

Reply:
In line 277, we have set the ready of the container to true, and the current test case has simulated the process of ready to change from true to false.
In addition, I think that readinessManager should not contain containers without readiness probes.

Reply:
> so , the following code wouldn't obtain the status from readinessmanager.

This is the behavior we are testing though isn't it? By adding a test we are ensuring future changes to this function does not retreive status from readiness manager if the container hasn't started yet.

Reply:
I think we can just adapt the test you added to include an entry in readiness manager instead of adding a new test.

Reply:
Yes, you are right. I should set up test cases based on problem scenarios, not the current code implementation.